### DISCUSSION TITLE
Start snow owl on a laptop with less 6gb  when it requires 6gb

### DISCUSSION THREAD

**Discussion by Cv66user** [b2ihealthcare/snow-owl#1042] [Q&A] [answered]
I have centos VM limited to 2gb  running on my win 10 hyper v, but my laptop physically has a max ram of 6gb. The container fails to start due to not having 6gb.  I tried docker run -m 1gb but it doesn't seem to recognize the parameter. I can see the xmx setting in the config file on a working server with 32gb. How can I start up snow owl with less than 6gb? 

I couldn't even do a docker cp to overwrite the config file as the container needs to be started in the first place

**Comment by cmark** [comment, marked as answer]
Hi @Cv66user,

Running both an Elasticsearch and Snow Owl with less than 6g of RAM (2g in this case) can be accomplished by changing the following lines in the official docker-compose.yml file:

Change the Elasticsearch JVM Heap setting to 1g here:
https://github.com/b2ihealthcare/snow-owl/blob/8.x/docker/docker-compose.yml#L13

And the Snow Owl heap setting to 1g here:
https://github.com/b2ihealthcare/snow-owl/blob/8.x/docker/docker-compose.yml#L41

Please note that running Snow Owl with this low amount of memory can only be used with a small dataset, like a [minified SNOMED CT dataset](https://github.com/b2ihealthcare/snow-owl/blob/8.x/tests/com.b2international.snowowl.test.commons/resources/snomed/SnomedCT_InternationalRF2_PRODUCTION_20210731T120000Z-minified.zip) and for testing purposes only (all Snow Owl test cases run with this 1g-1g setup on our CI infrastructure).

I hope this helps.

Cheers,
Mark

**Reply by Cv66user** [reply]
Thank you

**Comment by Cv66user** [comment]
Hi

Sorry to ask again, I am trying to run this on a Windows 2019 with 32GB ram with Docker Natively. I have not made the above changes.

Running A helloworld container is fine

But if I run: docker run --name A -p8080:8080 snowOwlContainerImage
I get: Native memory allocation (mmap) failed to map 6267994112 bytes for committing reserved memory, error='not enough space'

I tried: docker run --memory 25g --name B  -p8080:8080 snowOwlContainerImage
I get: Native memory allocation (mmap) failed to map 6267994112 bytes for committing reserved memory, error='not enough space'

I tried: docker run  --env SOJAVA_OPTS="-Xms1gb -Xmx20g" --memory 25g --name C -p8080:8080 snowOwlContainerImage
I get: Native memory allocation (mmap) failed to map 899284992 bytes for committing reserved memory, error='not enough space'

But this is a brand new VM with 32gb. Am I going to need 32gb + Windows Operating system RAM overhead?
How should the command be ran?


I am able to run the container if I use a 32GB VM with Docker Desktop but limit the desktop service to 15gb.

**Reply by cmark** [reply]
Hi @Cv66user,

Could you please try to use the following docker-compose configuration instead of trying to run the Snow Owl container directly? We don't have much experience with Windows Server 2019 and there might be some kind of system-wide configuration setting missing and would like to see if our official example works.

Example:
https://github.com/b2ihealthcare/snow-owl/tree/8.x/docker/compose

The error indicates that the Java Virtual Machine is unable to allocate the required memory. This has happened to me in Docker Desktop environments with the previous Hyper-V environments on my local dev env where I needed to increase the available resources. Not sure about the Docker service under Windows 2019, but if that's Hyper-V based then that might need to be increased there as well. If that is based on WSL2 then that manages the resources automatically.

I hope this helps.

Cheers,
Mark

**Reply by cmark** [reply]
Also, a `.wslconfig` configuration file can be placed in the current user's home directory to allow the WSL2 system to limit the memory available to docker containers.

```
[wsl2]
memory=3GB   # Limits VM memory in WSL 2 up to 3GB
```

**Reply by Cv66user** [reply]
Just tried

docker compose -f c:\temp\docker-compose.yaml
Unknown -f shorthand

docker compose
compose is not a docker command

Isnt Compose supported under Windows Docker?



Tried with 
c:\users\mypc\.wslconfig
[wsl]
memory=20gb

Then ran: docker run --name ContainerNameA -p 8080:8080 SnowOwlImage
I get: Native memory allocation (mmap) failed to map 6267994112 bytes for committing reserved memory, error='not enough space'


Even tried various switches such as 
docker run --env "ES_JAVA_OPTS=-Xms8gb -Xmx10g SO_JAVA_OPTS=-Xms6gb -Xmx6g"" --memory 15gb --name ContainerNameA -p 8080:8080 SnowOwlImage

With and without --memory --env switches gives: Native memory allocation (mmap) failed to map 6267994112 bytes for committing reserved memory, error='not enough space'

**Reply by cmark** [reply]
Do you have WSL1 or 2 installed on your Windows 2019 Server?

**Reply by Cv66user** [reply]
One I believe

**Reply by cmark** [reply]
Would it be possible to upgrade to Windows Server 2022? Or upgrade to WSL2? (internet is not that helpful and I'm not an expert of Windows Server either)

This is an older issue in a now archived repository, but it might have some workarounds that could help:
https://github.com/elastic/elasticsearch-docker/issues/43
